### PR TITLE
Feature/new sealing phase 2 b

### PR DESCRIPTION
**Pull request by arrivets**
### Sealing and Verification phase 2b

- re-request approvals
- emergency sealing
- limit number of receipts per block
- seal validation refactoring

### CODE REVIEW

File: integration/testnet/network.go [hunk 1]
```diff
@@ -499,7 +499,7 @@ func (net *FlowNetwork) AddNode(t *testing.T, bootstrapDir string, nodeConf Cont
 
 		nodeContainer.addFlag("rpc-addr", fmt.Sprintf("%s:9000", nodeContainer.Name()))
 		nodeContainer.bindPort(hostPort, containerPort)
-		nodeContainer.Ports[GhostNodeAPIPort] = hostPort
+
```
Comment (AlexHentschel) [inline integration/testnet/network.go:502]: I am wondering if this is the reason the integration test fails:
```
--- FAIL: TestExecutionFailingTxReverted (13.04s)
    --- FAIL: TestExecutionFailingTxReverted/TestExecutionFailingTxReverted (13.03s)
        suite.go:33: 
            	Error Trace:	suite.go:33
            	            				suite.go:115
            	            				suite.go:148
            	Error:      	Received unexpected error:
            	            	ghost node API port not found
            	Test:       	TestExecutionFailingTxReverted/TestExecutionFailingTxReverted
            	Messages:   	could not get ghost client
```

**Review by AlexHentschel — APPROVED** [review]
Looks good to me. We have accumulated some tech debt, but I think it it worth trying to push this to MainNet.

File: integration/testnet/network.go (hunk 1 above)
Comment (AlexHentschel) [reply integration/testnet/network.go]: reverting this change fixed the integration test 😅

**Review by durkmurder — APPROVED** [review]
Looks good

**Pull request merged**